### ISSUE TITLE
Unhandled exception first time a breakpoint is hit M26

### ISSUE THREAD

**Issue by ksdboasit** [flutter/flutter-intellij#2447]
The first time I hit a breakpoint during a debug run, I I get a whole lot of red text in the console. Sometimes the debug session is terminated, and other times it can be continued by hitting resume.

At first I did this on IntelliJ CE with iPhone simulator, but it also happened with any combination of Android Studio/IntelliJ and iPhone/Android emulator.

Ran flutter upgrade before any of this, and flutter doctor checked out fine (see below)
## Steps to Reproduce
1. Create new flutter project of type application.
2. Set a breakpoint at _counter++;
3. Start a debug session.

_Please tell us what you were doing and what went wrong_
Hit the + button on your emulator of choice and watch the console. I got the following:

```
Launching lib/main.dart on iPhone X in debug mode...
Starting Xcode build...
Xcode build done.
Syncing files to device iPhone X...
compiler message: Unhandled exception:
compiler message: NoSuchMethodError: The getter 'loader' was called on null.
compiler message: Receiver: null
compiler message: Tried calling: loader
compiler message: #0      Object.noSuchMethod (dart:core-patch/dart:core/object_patch.dart:46)
compiler message: #1      IncrementalCompiler.compileExpression.<anonymous closure> (package:front_end/src/fasta/incremental_compiler.dart:429)
compiler message: <asynchronous suspension>
compiler message: #2      CompilerContext.runInContext.<anonymous closure>.<anonymous closure> (package:front_end/src/fasta/compiler_context.dart:118)
compiler message: #3      new Future.sync (dart:async/future.dart:222)
compiler message: #4      CompilerContext.runInContext.<anonymous closure> (package:front_end/src/fasta/compiler_context.dart:118)
compiler message: #5      _rootRun (dart:async/zone.dart:1126)
compiler message: #6      _CustomZone.run (dart:async/zone.dart:1023)
compiler message: #7      _runZoned (dart:async/zone.dart:1518)
compiler message: #8      runZoned (dart:async/zone.dart:1465)
compiler message: #9      CompilerContext.runInContext (package:front_end/src/fasta/compiler_context.dart:117)
compiler message: #10     IncrementalCompiler.compileExpression (package:front_end/src/fasta/incremental_compiler.dart:428)
compiler message: <asynchronous suspension>
compiler message: #11     IncrementalCompiler.compileExpression (package:vm/incremental_compiler.dart:103)
compiler message: #12     FrontendCompiler.compileExpression (package:vm/frontend_server.dart:448)
compiler message: <asynchronous suspension>
compiler message: #13     _FlutterFrontendCompiler.compileExpression (package:frontend_server/server.dart:55)
compiler message: #14     listenAndCompile.<anonymous closure> (package:vm/frontend_server.dart:650)
compiler message: <asynchronous suspension>
compiler message: #15     _RootZone.runUnaryGuarded (dart:async/zone.dart:1316)
compiler message: #16     _BufferingStreamSubscription._sendData (dart:async/stream_impl.dart:336)
compiler message: #17     _BufferingStreamSubscription._add (dart:async/stream_impl.dart:263)
compiler message: #18     _SinkTransformerStreamSubscription._add (dart:async/stream_transformers.dart:68)
compiler message: #19     _EventSinkWrapper.add (dart:async/stream_transformers.dart:15)
compiler message: #20     _StringAdapterSink.add (dart:convert/string_conversion.dart:268)
compiler message: #21     _LineSplitterSink._addLines (dart:convert/line_splitter.dart:154)
compiler message: #22     _LineSplitterSink.addSlice (dart:convert/line_splitter.dart:129)
compiler message: #23     StringConversionSinkMixin.add (dart:convert/string_conversion.dart:189)
compiler message: #24     _SinkTransformerStreamSubscription._handleData (dart:async/stream_transformers.dart:120)
compiler message: #25     _RootZone.runUnaryGuarded (dart:async/zone.dart:1316)
compiler message: #26     _BufferingStreamSubscription._sendData (dart:async/stream_impl.dart:336)
compiler message: #27     _BufferingStreamSubscription._add (dart:async/stream_impl.dart:263)
compiler message: #28     _SinkTransformerStreamSubscription._add (dart:async/stream_transformers.dart:68)
compiler message: #29     _EventSinkWrapper.add (dart:async/stream_transformers.dart:15)
compiler message: #30     _StringAdapterSink.add (dart:convert/string_conversion.dart:268)
compiler message: #31     _StringAdapterSink.addSlice (dart:convert/string_conversion.dart:273)
compiler message: #32     _Utf8ConversionSink.addSlice (dart:convert/string_conversion.dart:348)
compiler message: #33     _Utf8ConversionSink.add (dart:convert/string_conversion.dart:341)
compiler message: #34     _ConverterStreamEventSink.add (dart:convert/chunked_conversion.dart:86)
compiler message: #35     _SinkTransformerStreamSubscription._handleData (dart:async/stream_transformers.dart:120)
compiler message: #36     _RootZone.runUnaryGuarded (dart:async/zone.dart:1316)
compiler message: #37     _BufferingStreamSubscription._sendData (dart:async/stream_impl.dart:336)
compiler message: #38     _BufferingStreamSubscription._add (dart:async/stream_impl.dart:263)
compiler message: #39     _StreamController&&_SyncStreamControllerDispatch._sendData (dart:async/stream_controller.dart:763)
compiler message: #40     _StreamController._add (dart:async/stream_controller.dart:639)
compiler message: #41     _StreamController.add (dart:async/stream_controller.dart:585)
compiler message: #42     _Socket._onData (dart:io-patch/socket_patch.dart:1672)
compiler message: #43     _RootZone.runUnaryGuarded (dart:async/zone.dart:1316)
compiler message: #44     _BufferingStreamSubscription._sendData (dart:async/stream_impl.dart:336)
compiler message: #45     _BufferingStreamSubscription._add (dart:async/stream_impl.dart:263)
compiler message: #46     _StreamController&&_SyncStreamControllerDispatch._sendData (dart:async/stream_controller.dart:763)
compiler message: #47     _StreamController._add (dart:async/stream_controller.dart:639)
compiler message: #48     _StreamController.add (dart:async/stream_controller.dart:585)
compiler message: #49     new _RawSocket.<anonymous closure> (dart:io-patch/socket_patch.dart:1245)
compiler message: #50     _NativeSocket.issueReadEvent.issue (dart:io-patch/socket_patch.dart:797)
compiler message: #51     _microtaskLoop (dart:async/schedule_microtask.dart:41)
compiler message: #52     _startMicrotaskLoop (dart:async/schedule_microtask.dart:50)
compiler message: #53     _runPendingImmediateCallback (dart:isolate-patch/dart:isolate/isolate_patch.dart:113)
compiler message: #54     _RawReceivePortImpl._handleMessage (dart:isolate-patch/dart:isolate/isolate_patch.dart:166)
```

A couple of times I also got:
```
SocketException: Write failed (OS Error: Broken pipe, errno = 32), port = 0
#0      _rootHandleUncaughtError.<anonymous closure> (dart:async/zone.dart:1114)
#1      _microtaskLoop (dart:async/schedule_microtask.dart:41)
#2      _startMicrotaskLoop (dart:async/schedule_microtask.dart:50)
#3      _runPendingImmediateCallback (dart:isolate-patch/dart:isolate/isolate_patch.dart:113)
#4      _RawReceivePortImpl._handleMessage (dart:isolate-patch/dart:isolate/isolate_patch.dart:166)
```

When this occurred the debug session was also terminated.

## Version info

[✓] Flutter (Channel master, v0.5.6-pre.84, on Mac OS X 10.13.4 17E199, locale en-DK)
    • Flutter version 0.5.6-pre.84 at /Users/kds/flutterTest/flutter
    • Framework revision 0fb209726f (21 hours ago), 2018-06-27 10:57:34 +0100
    • Engine revision 6fe748490d
    • Dart version 2.0.0-dev.63.0.flutter-4c9689c1d2

[✓] Android toolchain - develop for Android devices (Android SDK 27.0.3)
    • Android SDK at /Users/kds/Library/Android/sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-27, build-tools 27.0.3
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1024-b01)
    • All Android licenses accepted.

[✓] iOS toolchain - develop for iOS devices (Xcode 9.4)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 9.4, Build version 9F1027a
    • ios-deploy 1.9.2
    • CocoaPods version 1.5.3

[✓] Android Studio
    • Android Studio at /Applications/Android Studio 3.2 Preview.app/Contents
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
    ✗ Dart plugin not installed; this adds Dart specific functionality.
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1136-b04)

[✓] Android Studio (version 3.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 26.0.1
    • Dart plugin version 173.4700
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1024-b01)

[✓] IntelliJ IDEA Community Edition (version 2018.1.5)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 26.0.2
    • Dart plugin version 181.4892.1

[✓] Connected devices (1 available)
    • iPhone X • 4BFF15F8-BECF-4EA3-B6AC-815A616B8B29 • ios • iOS 11.4 (simulator)

• No issues found!

**Comment by devoncarew** [comment]
@ksdboasit, thanks for the report! Debugging has several issues right now when used against the flutter master channel. This PR: https://github.com/flutter/flutter/pull/18657 will work around the biggest issue - what I think you're hitting.

**Comment by devoncarew** [comment]
I can confirm that this is fixed with flutter/flutter#18657 - you can see the fix if you run `fluter upgrade`.
